src/okta/createOktaClient.ts

```diff
@@ -35,9 +35,9 @@ export class RequestExecutorWithEarlyRateLimiting extends okta.RequestExecutor {
 
   async fetch(request: any) {
     const now = Date.now();
-    if (this.requestAfter && this.requestAfter > now) {
-      const delayMs = this.requestAfter - now;
-      await new Promise((resolve) => setTimeout(resolve, delayMs));
+    if (this.getThrottleActivated()) {
+      const delayMs = this.requestAfter! - now;
```

Comment:
I believe that `getThrottleActivated()` is responsible for setting `this.requestAfter`, but I think it would be best/most safe to make a check:

```suggestion
    if (this.getThrottleActivated() && this.requestAfter) {
      const delayMs = this.requestAfter - now;
```

Generally, a `!` on our own code is suspicious to me. I've grown accustomed to it on API response data structures because most generated types from the libraries will type properties as optional even though it would make no sense at all for those properties to be `undefined` for REST responses.

Reply:
Okay, cool. Done.